python/pyarrow/types.pxi

```diff
@@ -635,6 +635,30 @@ cdef class Schema:
 
         return pyarrow_wrap_schema(new_schema)
 
+    def set(self, int i, Field field):
```

Comment:
`set_field` for nominal consistency?

Reply:
I have sticked to `set` as we also have simply `remove` and `insert` on the Schema.

Reply:
:+1: